bloom-housing/bloom
# feat: adding preference sheets to lottery export

This PR addresses #4056 


## Description
This adds the preference sheets to the lottery export as well as a few other things:

- preference sheets
    - added
- demographics
    - now based on role just like application csv export
- columns
    - added preference rank
    - still need to double check column order
- permissions
    - UI is now hooked up to do a partner export

## How Can This Be Tested/Reviewed?
- head to the partners site
- on a closed listing with applications and preferences on the listing (and to make it easier claimed preferences on at least some of the applications)
- generate the lottery results
- get the spreadsheet export

you should notice individual sheets for each preference as well as the original raw sheet

## Author Checklist:


## Review Process:

- Read and understand the issue
- Ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Either (1) explicitly ask a clarifying question, (2) request changes, or (3) approve the PR, even if there are very small remaining changes, if you don't need to re-review after the updates
